src/webgpu/shader/validation/parse/align.spec.ts

```diff
@@ -61,3 +61,112 @@ fn main() -> @location(0) vec4<f32> {
 }`;
     t.expectCompileResult(kValidAlign.has(t.params.align), code);
   });
+
+g.test('align_required_alignment')
+  .desc('Test that the align with an invalid size is an error')
+  .params(u =>
+    u
+      .combine('address_space', ['storage', 'uniform'])
+      // These test a few cases:
+      //  * 1 -- Invalid, alignment smaller then all the required alignments
+      //  * alignment -- Valid, the required alignment
+      //  * 32 -- Valid, an alignment larger then the required alignment.
+      .combine('align', [1, 'alignment', 32])
+      .combine('type', [
+        { name: 'i32', storage: 4, uniform: 4 },
+        { name: 'u32', storage: 4, uniform: 4 },
+        { name: 'f32', storage: 4, uniform: 4 },
+        { name: 'f16', storage: 2, uniform: 2 },
+        { name: 'atomic<i32>', storage: 4, uniform: 4 },
+        { name: 'vec2<i32>', storage: 8, uniform: 8 },
+        { name: 'vec2<f16>', storage: 4, uniform: 4 },
+        { name: 'vec3<u32>', storage: 16, uniform: 16 },
+        { name: 'vec3<f16>', storage: 8, uniform: 8 },
+        { name: 'vec4<f32>', storage: 16, uniform: 16 },
+        { name: 'vec4<f16>', storage: 8, uniform: 8 },
+        { name: 'mat2x2<f32>', storage: 8, uniform: 8 },
+        { name: 'mat3x2<f32>', storage: 8, uniform: 8 },
+        { name: 'mat4x2<f32>', storage: 8, uniform: 8 },
+        { name: 'mat2x2<f16>', storage: 4, uniform: 4 },
+        { name: 'mat3x2<f16>', storage: 4, uniform: 4 },
+        { name: 'mat4x2<f16>', storage: 4, uniform: 4 },
+        { name: 'mat2x3<f32>', storage: 16, uniform: 16 },
+        { name: 'mat3x3<f32>', storage: 16, uniform: 16 },
+        { name: 'mat4x3<f32>', storage: 16, uniform: 16 },
+        { name: 'mat2x3<f16>', storage: 8, uniform: 8 },
+        { name: 'mat3x3<f16>', storage: 8, uniform: 8 },
+        { name: 'mat4x3<f16>', storage: 8, uniform: 8 },
+        { name: 'mat2x4<f32>', storage: 16, uniform: 16 },
+        { name: 'mat3x4<f32>', storage: 16, uniform: 16 },
+        { name: 'mat4x4<f32>', storage: 16, uniform: 16 },
+        { name: 'mat2x4<f16>', storage: 8, uniform: 8 },
+        { name: 'mat3x4<f16>', storage: 8, uniform: 8 },
+        { name: 'mat4x4<f16>', storage: 8, uniform: 8 },
+        // The 0 for uniform means we'll skip this case as the `vec2` as an array element is not
+        // valid in uniform address space
+        { name: 'array<vec2<i32>, 2>', storage: 8, uniform: 0 },
+        { name: 'array<vec4<i32>, 2>', storage: 8, uniform: 16 },
+        { name: 'S', storage: 8, uniform: 16 },
+      ])
+      .beginSubcases()
+  )
+  .beforeAllSubcases(t => {
+    if (t.params.type.name.includes('f16')) {
+      t.selectDeviceOrSkipTestCase('shader-f16');
+    }
+  })
+  .fn(t => {
+    if (t.params.address_space === 'uniform' && t.params.type.name.startsWith('atomic')) {
+      t.skip('No atomics in uniform address space');
```

Comment:
Are these skips instead of expecting to not compile because they are invalid due to non-align rules reasons? If so could you put a comment mentioning that?

Reply:
Added a comment for one, and removed the other. I expanded the failure cases to cover the second one below and added comments there as well.